### PR TITLE
Docs on linting

### PR DESCRIPTION
**Pull request by timja**
See https://github.com/jenkinsci/jenkins/pull/6863#issuecomment-1252978467

An option (on top of this) is https://github.com/jenkinsci/jenkins/pull/7075 which makes lint fixing for prettier take ~100ms instead of 7 seconds, as it only runs on changed files

### Proposed changelog entries

- (skip)
- ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  - Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7153"></a>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -74,6 +74,24 @@ You can use IntelliJ IDEA (preferred) or VS Code (alternate) in the browser.
...
 which will open the workspace in IntelliJ IDEA using JetBrains Gateway.
 
+### Linting
+
+For linting we use a number of tools:
+
+- eslint
+- stylelint
+- prettier
+- spotless
```
Comment (NotMyFault) [inline CONTRIBUTING.md:84]:
```suggestion
- [eslint](https://eslint.org/)
- [stylelint](https://stylelint.io/)
- [prettier](https://prettier.io/)
- [spotless](https://github.com/diffplug/spotless)
```
Any objections about linking the tools? I doubt, everyone knows all linters, especially the front end ones.

Comment (timja) [reply CONTRIBUTING.md:84]: sure I thought about it but wasn't sure, happy to link

**Review by daniel-beck — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Comment by MarkEWaite** [comment]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

**Review by basil — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -74,6 +74,24 @@ You can use IntelliJ IDEA (preferred) or VS Code (alternate) in the browser.
 If you prefer using IntelliJ IDEA, you can setup Gitpod integration with JetBrains Gateway using the instructions on [gitpod.io](https://www.gitpod.io/docs/ides-and-editors/intellij),
 which will open the workspace in IntelliJ IDEA using JetBrains Gateway.
 
+### Linting
+
+For linting we use a number of tools:
```
Comment (jglick) [inline CONTRIBUTING.md:79]: What about Checkstyle? This has no automatic fix right?

**Review by jglick — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -74,6 +74,24 @@ You can use IntelliJ IDEA (preferred) or VS Code (alternate) in the browser.
...
+
+- [eslint](https://eslint.org/)
+- [stylelint](https://stylelint.io/)
+- [prettier](https://prettier.io/)
+- [spotless](https://github.com/diffplug/spotless)
```
Comment (timja) [inline CONTRIBUTING.md:84]:
```suggestion
- [checkstyle](https://checkstyle.sourceforge.io/)
- [eslint](https://eslint.org/)
- [prettier](https://prettier.io/)
- [spotless](https://github.com/diffplug/spotless)
- [stylelint](https://stylelint.io/)

```

File: CONTRIBUTING.md (hunk 2 above)
Comment (timja) [reply CONTRIBUTING.md:79]: I forgot that one. I tried to cover it with "To automatically fix most issues run"

A number of eslint and stylelint issues are not automatically fixable (although many are)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -74,6 +74,26 @@ You can use IntelliJ IDEA (preferred) or VS Code (alternate) in the browser.
...
+### Linting
+
+For linting we use a number of tools:
+
+- [checkstyle](https://checkstyle.sourceforge.io/)
+- [eslint](https://eslint.org/)
+- [prettier](https://prettier.io/)
+- [spotless](https://github.com/diffplug/spotless)
+- [stylelint](https://stylelint.io/)
+
```
Comment (timja) [inline CONTRIBUTING.md:86]:
```suggestion
```

**Pull request merged**